src/plugins/cpptemplate/cpptemplate_delegate.hpp

```diff
@@ -0,0 +1,29 @@
+/**
+ * @file
+ *
+ * @brief Delegate definitions for the `cpptemplate` plugin
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#ifndef ELEKTRA_CPP_TEMPLATE_DELEGATE_HPP
+#define ELEKTRA_CPP_TEMPLATE_DELEGATE_HPP
+
+#include <kdberrors.h>
+#include <kdbplugin.hpp>
+
+namespace elektra
+{
+
+class CppTemplateDelegate
+{
+	using CppKeySet = kdb::KeySet;
+
+public:
```

Comment:
open, close, ... not declared?

Reply:
We do not need them here. We can add any function we want to the delegate. I feel a little silly telling  you that though, after all you wrote [`kdbplugin.hpp`](https://github.com/ElektraInitiative/libelektra/blob/master/src/bindings/cpp/include/kdbplugin.hpp). I feel like, I might not understand the whole delegator/delegate approach after all…

Reply:
Sorry for the confusion. I simply forgot what I wrote (and did not see that you include kdbplugin.hpp) and thought that the cpptemplate_delegate.hpp fully implements the delegator.

It makes a lot of sense how it is. You should only mention that the constructor/destructor should contain the open/close functionality.